src/main/java/com/example/messageserver/kafka/KafkaListeners.java

```diff
@@ -23,10 +23,12 @@ public class KafkaListeners {
     private CountDownLatch latch = new CountDownLatch(1);
 
     @KafkaListener(topics = orderTopic, groupId = "name", errorHandler = "priceErrorHandler")
-    public void consumeMessage(BatteryStatisticDto batteryDto) throws JsonProcessingException {
-        if(batteryDto.getTotalWattCapacity()>50.0){
-           throw new IllegalArgumentException("Too much watt capacity " + batteryDto.getTotalWattCapacity());
+    public void consumeMessage(BatteryStatisticsDtoEvent batteryStatisticsDtoEvent) throws JsonProcessingException {
```

Comment:
Remove throws clause, since deserializing is done before consuming the message now.